### ISSUE TITLE
[Bug] “The character encoding was not declared”

### ISSUE THREAD

**Issue by j9t** [theoludwig/html-w3c-validator#8]
Follow-up for #6:

## Steps to reproduce

1. Locally validate [the following HTML](https://mirrors.meiert.org/havewestoppedkillingyet.org/)

(Settings:)

```
{
  "$schema": "https://raw.githubusercontent.com/theoludwig/html-w3c-validator/master/schema/html-w3c-validatorrc-schema.json",
  "files": ["./index.html"],
  "severities": ["info", "warning", "error"]
}
```

## The current behavior

1.  `error   │ The character encoding was not declared. Proceeding using “windows-1252”.`

## The expected behavior

1. No errors

As mentioned in #6:

The document in question uses UTF-8, which is [also defined by HTTP header](https://redbot.org/?uri=https%3A%2F%2Fmirrors.meiert.org%2Fhavewestoppedkillingyet.org%2F).

HTML uses [UTF-8 as the default encoding](https://html.spec.whatwg.org/multipage/semantics.html#charset).

There is no requirement to declare the encoding in the file itself.

As such this shouldn’t be an error, and the fallback should probably be UTF-8.

**Comment by theoludwig** [comment]
This error is returned from the "[Nu Html Checker](https://validator.w3.org/nu/)", so it's not an issue from this tool, it's basically doing an API call, and format in a nice way for CLI usage.
https://github.com/theoludwig/html-w3c-validator/blob/fbe3830db7522056a425d22aeb8cfe0d274812bb/src/validateHTML.ts#L23

**Comment by j9t** [comment]
I’d like to kindly challenge that, in the light of making html-w3c-validator better.

For a rationale why that error is bogus, see the original issue report. The error makes no sense, neither from a declaration standpoint (the encoding is declared) nor from a fallback standpoint (which should be UTF-8).

Then, the issue is [not being reported in the web-based validator](https://validator.nu/?doc=https%3A%2F%2Fmirrors.meiert.org%2Fhavewestoppedkillingyet.org%2F).

I can imagine there’s little appetite to fix something wrong somewhere in the Nu validator implementation. But as this only surfaces in html-w3c-validator, this may be good to either make sure is being fixed on the Nu validator end, or is being caught with html-w3c-validator.

**Comment by theoludwig** [comment]
> Then, the issue is [not being reported in the web-based validator](https://validator.nu/?doc=https%3A%2F%2Fmirrors.meiert.org%2Fhavewestoppedkillingyet.org%2F).

Exact, in the web based version, the error is not reported. Is there any reason? I would say if we find a reason, and we find it compeling, we might effectively do the same thing, ignore this error.

I'm not strongly against ignoring it, but IMO, this error can be easily resolved by adding `<meta charset="UTF-8" />` in the HTML (I can understand that with the HTTP Header and all that, it can be fixed, but still, I still feel like adding `<meta charset="UTF-8" />`  is not too much, and fix the issue as well).
